GUARD AH OT-insert wild puVar7 (local_14)

(rationale: see preceding patch in the series)

--- a/dd2.c
+++ b/dd2.c
@@ -10490,9 +10490,11 @@
         puVar7 = (undefined4 *)
                  (((int)((__otz + (__otz >> 0x1f) * -4) - (uint)((__otz >> 0x1f) << 1 < 0)) >> 2) *
                   4 + iVar4);
+        if ((uintptr_t)puVar7 >= 0x400000u && (uintptr_t)puVar7 < 0x900000u) {
         uVar2 = *puVar7;
         *puVar7 = local_14;
         *local_14 = uVar2;
+        }
         __rgb0 = (uint)*(byte *)(local_18 + 4);
         __vr3 = __vr2;
         _DAT_0074c4f4 = _DAT_0074c4e4;
